src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -56,66 +55,53 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
     }
 
     public async afterResolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<void> {
-        const { task, promise } = netSdkRunTaskProvider.createNetSdkRunTask(
-            {
-                netCore: {
-                    appProject: await this.inferProjPath(context.actionContext, context.folder),
-                }
+        const projectInfo = await this.getProjectInfo(debugConfiguration);
+        const runDefinition: Omit<NetSdkRunTaskDefinition, "type"> = {
+            netCore: {
+                appProject: await this.inferProjPath(undefined, debugConfiguration.netCore),
+            },
+            dockerRun: {
+                containerName: projectInfo[5]
             }
-        );
+        };
+
+        const { task, promise } = netSdkRunTaskProvider.createNetSdkRunTask(runDefinition);
 
         await tasks.executeTask(task);
 
         try {
             await promise;
         } catch (error) {
-            await NetSdkDebugHelper.clearWorkspaceState();
+            await ext.context.workspaceState.update(NetContainerBuildOptionsKey, '');
```

Comment:
We shouldn't need to set workspace state at all anymore, should we?

Reply:
This is for clearing the cache for "Choose .NET SDK Build vs Dockerfile build." We still need to clear the cache if user runs into an error with the  .NET SDK Build  and wants to use the Dockerfile build

Reply:
One of these days I'll remember that; do consider overriding resolveDebugConfiguration in NetSdkDebugHelper to avoid having to sprinkle these calls to reset the value throughout the debug provider logic. This does assume that everywhere you're currently resetting the state you're also throwing an error:

```
public override async resolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<ResolvedDebugConfiguration | undefined> {
    try {
        return await super.resolveDebugConfiguration(context, debugConfiguration);
    } catch (error) {
        await ext.context.workspaceState.update(NetcontainerBuildOptionsKey, '');
        throw error;
    }
}
```